### DISCUSSION TITLE
api/hello read as query {api and hello} in page dir

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [vercel/next.js#60700] [Help]
### Summary

I have dyanmaic route /[userId]/[postId] in page directory.
but when I access path that is not exist Ex: api/hello
Then it become query for dynmic route .
Is there any solution ?
I don't want to dynamic url like this  user/id/post/id

![image](https://github.com/vercel/next.js/assets/71011043/d0e84ed7-3aac-4742-afaf-07b4c72abd87)

**Comment by icyJoseph** [comment]
If there's no `/api/hello` explicitly defined in your directory, then Next.js does a fallback onto the catch all route, `[userId]/[postId]`.

You could use next.js rewrites, to match `/api/<something>`, and rewrite that to an API route.

**Reply by thanhtutzaw** [reply]
I don't want to rewrite but I want to tell that is wrong api error message . How can I handle it ? Or we don't have to hadle it ?

**Reply by joulev** [reply]
@thanhtutzaw You can check inside the `/[userId]/[postId]` path like

```tsx
if (params.userId === "api") notFound()
```

As Joseph said, this is by design, since nothing prevents `userId` to be `"api"` unless you actively enforce it (e.g. with the `if` statement above).
